js/digifinex.js

```diff
@@ -0,0 +1,515 @@
+            const precision = {
+                'price': market[1],
+                'amount': market[0],
+            };
+            const limits = {
+                'amount': {
+                    'min': market[2],
+                    'max': undefined,
+                },
+                'price': {
+                    'min': undefined,
+                    'max': undefined,
+                },
+                'cost': {
+                    'min': market[3],
+                    'max': undefined,
+                },
+            };
+            result.push ({
+                'id': id,
+                'symbol': symbol,
+                'base': base,
+                'quote': quote,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'active': true,
+                'precision': precision,
+                'limits': limits,
+                'info': market,
+            });
+        }
+        return result;
+    }
+
+    async fetchBalance (params = {}) {
+        const response = await this.privateGetMyposition ();
+        const result = { 'info': response };
+        const free = response['free'];
+        const frozen = response['frozen'];
+        const keysFree = Object.keys (free);
+        const keysFrozen = Object.keys (frozen);
+        const keys = this.arrayConcat (keysFree, keysFrozen);
+        for (let i = 0; i < keys.length; i++) {
+            const uppercase = keys[i].toUpperCase ();
+            const account = this.account ();
+            if (this.inArray (keys[i], keysFree)) {
+                account['free'] = free[keys[i]];
+            }
+            if (this.inArray (keys[i], keysFrozen)) {
+                account['used'] = frozen[keys[i]];
+            }
+            account['total'] = this.sum (account['free'] + account['used']);
+            result[uppercase] = account;
+        }
+        return this.parseBalance (result);
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        // limit can be up to 100, defaults to 10
+        const markets = await this.loadMarkets ();
+        const market = markets[symbol];
+        symbol = market['baseId'] + '_' + market['quoteId'];
+        const request = {
+            'market': symbol,
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        const response = await this.publicV3GetOrderBook (this.extend (request, params));
+        return this.parseOrderBook (response, parseInt (response['date']) * 1000);
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        if (symbols !== undefined) {
+            throw new BadRequest ('Only supports fetching all tickers');
+        }
+        const response = await this.publicGetTicker ();
+        const result = {};
+        const keys = Object.keys (response['ticker']);
+        for (let i = 0; i < keys.length; i++) {
+            let symbol = keys[i].toUpperCase ();
```

Comment:
In this case another symbol-related rule is volated: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#parsing-symbols. There are other similar lines throughout the file.